includes/wikia/services/DataMartService.class.php

```diff
@@ -88,49 +98,32 @@ protected static function getPageviewsForWikis ( $periodId, $wikis, $startDate,
 			}
 		}
 
-		$db = DataMartService::getDB();
-		$pageviews = ( new WikiaSQL() )->skipIf( self::isDisabled() )
-			->cacheGlobal( self::TTL )
-			->SELECT( 'wiki_id' )
-				->FIELD( "date_format(time_id,'%Y-%m-%d')" )->AS_( 'date' )
-				->FIELD( 'pageviews' )->AS_( 'cnt' )
-			->FROM( 'rollup_wiki_pageviews' )
-			->WHERE( 'period_id' )->EQUAL_TO( $periodId )
-				->AND_( 'wiki_id' )->IN( $wikis )
-				->AND_( 'time_id' )->BETWEEN( $startDate, $endDate )
-			->runLoop( $db, function ( &$pageViews, $row ) {
-				$pageViews[$row->wiki_id][$row->date] = $row->cnt;
-				$pageViews[$row->wiki_id]['SUM'] += $row->cnt;
-			} );
-
-		return $pageviews;
-	}
-
-	/**
-	 * get pageviews
-	 * @param array $dates ( YYYY-MM-DD, YYYY-MM-DD ... )
-	 * @return array $pageviews [ array( 'YYYY-MM-DD' => pageviews ) ]
-	 */
-	public static function getSumPageviewsMonthly ( $dates = array() ) {
-		$periodId = self::PERIOD_ID_MONTHLY;
+		try {
+			$db = DataMartService::getDB();
+			$pageviews =
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$pageviews". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)